x/merkledb/db.go

```diff
@@ -852,19 +875,12 @@ func (db *merkleDB) commitChanges(ctx context.Context, trieToCommit *trieView) e
 		return errNoNewRoot
 	}
 
-	// commit any outstanding cache evicted nodes.
-	// Note that we do this here because below we may Abort
-	// [db.nodeDB], which would cause us to lose these changes.
-	if err := db.nodeDB.Commit(); err != nil {
```

Comment:
shouldn't we keep `db.commit()` here for the reason that is mentioned in this comment?

Reply:
No longer happens with this change.  evictions write directly to the db now